src/sidebar/zeplinComponent/data/CumulativeBarrelDetailsStore.ts

```diff
@@ -0,0 +1,52 @@
+import BarrelDetails from "../../../common/domain/zeplinComponent/model/BarrelDetails";
+import Store from "../../../common/domain/store/Store";
+import BarrelError from "../../../common/domain/zeplinComponent/model/BarrelError";
+import BarrelType from "../../../common/domain/barrel/BarrelType";
+import Result from "../../../common/domain/store/Result";
+import BarrelDetailsStoreProvider from "../../../common/domain/zeplinComponent/data/BarrelDetailsStoreProvider";
+import { convertBarrelResult } from "../../../common/domain/barrel/util/barrelUtil";
+
+export default class CumulativeBarrelDetailsStore implements Store<BarrelDetails[], BarrelError> {
+    public constructor(private barrelId: string, private barrelType: BarrelType) { }
+
+    public get = async (): Promise<Result<BarrelDetails[], BarrelError>> => {
+        const leafId = this.barrelId;
+        const leafType = this.barrelType;
+        let currentId: string | undefined = leafId;
+        let currentType = leafType;
+        const accumulatedResult: Result<BarrelDetails[], BarrelError> = {
+            data: [],
+            errors: []
+        };
+
+        while (currentId) {
+            const childId: string | undefined = currentId === leafId ? undefined : leafId;
+            const childType: BarrelType | undefined = currentId === leafId ? undefined : leafType;
+
+            const { data, errors }: Result<BarrelDetails, BarrelError> = await BarrelDetailsStoreProvider
+                .get(currentId, currentType, childId, childType)
+                .get()
```

Comment:
This `get` looks `redundant`. Can we adjust `BarrelDetailsStoreProvider.get()` to do this?

Reply:
because we use a storeprovider here, first get gets store, second get gets store result

Reply:
Yeah, I checked that. I would expect first `get` to do all of this 😄

Reply:
Why do you expect that. You should expect a get call on XProvider to provide with an X, not a Y

Reply:
OK, I got it after investigating a bit more. So we return stores from providers and return results from stores. I was thinking directly returning results from providers but it was wrong 👍